typescript/sdk/src/providers/transactions/submitter/builder/TxSubmitterBuilder.ts

```diff
@@ -0,0 +1,163 @@
+          this.multiProvider,
+          txSubmitterOrType.chain,
+          txSubmitterOrType.impersonatedAccountTxSubmitterProps,
+        );
+        return this;
+      case TxSubmitterType.GNOSIS_SAFE:
+        assert(
+          txSubmitterOrType.gnosisSafeTxSubmitterProps,
+          'Must provide required props for Gnosis safe submitter.',
+        );
+        this.currentSubmitter = new GnosisSafeTxSubmitter<HTX, HTR>(
+          this.multiProvider,
+          txSubmitterOrType.chain,
+          txSubmitterOrType.gnosisSafeTxSubmitterProps,
+        );
+        return this;
+      default:
+        this.currentSubmitter = txSubmitterOrType.submitter;
+        return this;
+    }
+  }
+
+  /**
+   * Sets the current transformer for the builder.
+   * @param txTransformerOrType The transformer to add to the builder
+   */
+  public transform(
+    txTransformerOrType: TxTransformerXORType<HTX>,
+  ): TxSubmitterBuilder<HTX, HTR> {
+    assert(
+      this.currentSubmitter,
+      'No submitter specified for which to execute the transform.',
+    );
+
+    switch (txTransformerOrType.type) {
+      case TxTransformerType.ICA:
+        assert(
+          txTransformerOrType.interchainAccountTxTransformerProps,
+          'Must provide required props for interchain account submitter.',
+        );
+        this.currentTransformer = new InterchainAccountTxTransformer<HTX>(
+          this.multiProvider,
+          this.currentSubmitter.chain,
+          txTransformerOrType.chain,
+          txTransformerOrType.interchainAccountTxTransformerProps,
+        );
+        return this;
+      default:
+        this.currentTransformer = txTransformerOrType.transformer;
+        return this;
+    }
+  }
+
+  /**
+   * Submits a set of transactions to the builder.
+   * @param hyperlaneTxs The transactions to submit
+   */
+  public async submit(hyperlaneTxs: HTX[]): Promise<HTR[]> {
+    assert(
+      this.currentSubmitter,
+      'Must specify submitter to submit transactions.',
+    );
+
+    this.logger.info(
+      `Submitting ${hyperlaneTxs.length} transactions to the ${this.currentSubmitter.txSubmitterType} submitter...`,
+    );
+
+    let transformedHyperlaneTxs = hyperlaneTxs;
+    if (this.currentTransformer) {
+      transformedHyperlaneTxs = await this.currentTransformer.transformTxs(
+        hyperlaneTxs,
+        this.currentSubmitter.txSubmitterType,
+      );
+      this.logger.info(
+        `🔄 Transformed ${transformedHyperlaneTxs.length} transactions with the ${this.currentTransformer.txTransformerType} transformer...`,
+      );
+    }
+
+    const hyperlaneTxReceipts = await this.currentSubmitter.submitTxs(
+      transformedHyperlaneTxs,
+    );
+    this.logger.info(
+      `✅ Successfully submitted ${hyperlaneTxReceipts.length} transactions to the ${this.currentSubmitter.txSubmitterType} submitter.`,
+    );
+
+    this.currentSubmitter = undefined;
+    this.currentTransformer = undefined;
```

Comment:
calling `submitTxs` twice would have quite different behavior with this implementation

Reply:
This was with the assumption we would only transform once at a time. We now support multiple transforms for a single submit operation